src/BenchmarkDotNet.Analyzers/AnalyzerHelper.cs

```diff
@@ -212,19 +207,16 @@ private static bool IsAssignableTo(string codeTemplate1, string codeTemplate2, C
             return false;
         }
 
-        return HasNoCompilerDiagnostics(string.Format(codeTemplate2, usingDirective, targetType, valueType, constantLiteral), compilation, languageVersion);
+        return HasNoCompilerDiagnostics(string.Format(codeTemplate2, targetType, valueType, constantLiteral), compilation);
     }
 
-    private static bool HasNoCompilerDiagnostics(string code, Compilation compilation, LanguageVersion languageVersion)
+    private static bool HasNoCompilerDiagnostics(string code, Compilation compilation)
     {
-        var compilationTestSyntaxTree = CSharpSyntaxTree.ParseText(code, new CSharpParseOptions(languageVersion));
-
-        var syntaxTreesWithInterceptorsNamespaces = compilation.SyntaxTrees.Where(st => st.Options.Features.ContainsKey(InterceptorsNamespaces));
+        var syntaxTree = CSharpSyntaxTree.ParseText(code);
 
         var compilerDiagnostics = compilation

```

Comment:
How do you cirmcumvent InterceptorsNamespaces here?

Reply:
`compilation.ClassifyConversion` and `semanticModel.ClassifyConversion` just does the thing without having to manually construct a dummy syntax tree.